src/contracts/pods/EigenPod.sol

```diff
@@ -169,133 +140,114 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         emit RestakingActivated(podOwner);
     }
 
+    /*******************************************************************************
+                                    EXTERNAL METHODS
+    *******************************************************************************/
+
     /// @notice payable fallback function that receives ether deposited to the eigenpods contract
     receive() external payable {
         nonBeaconChainETHBalanceWei += msg.value;
         emit NonBeaconChainETHReceived(msg.value);
     }
 
     /**
-     * @notice This function records an update (either increase or decrease) in a validator's balance.
-     * @param oracleTimestamp The oracleTimestamp whose state root the proof will be proven against.
-     *        Must be within `VERIFY_BALANCE_UPDATE_WINDOW_SECONDS` of the current block.
-     * @param validatorIndices is the list of indices of the validators being proven, refer to consensus specs 
-     * @param stateRootProof proves a `beaconStateRoot` against a block root fetched from the oracle
-     * @param validatorFieldsProofs proofs against the `beaconStateRoot` for each validator in `validatorFields`
-     * @param validatorFields are the fields of the "Validator Container", refer to consensus specs
-     * @dev For more details on the Beacon Chain spec, see: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
+     * @dev Create a checkpoint used to prove this pod's active validator set. Checkpoints are completed 
+     * by submitting one checkpoint proof per ACTIVE validator. During the checkpoint process, the total 
+     * change in ACTIVE validator balance is tracked, and any validators with 0 balance are marked `WITHDRAWN`.
+     * @dev Once finalized, the pod owner is awarded shares corresponding to:
+     * - the total change in their ACTIVE validator balances
+     * - any ETH in the pod not already awarded shares.
+     * @dev A checkpoint cannot be created if the pod already has an outstanding checkpoint. If
+     * this is the case, the pod owner MUST complete the existing checkpoint before starting a new one.
```

Comment:
what would be the side effects of allowing cancelling a checkpoint and starting a new one?
is there intermediate storage that would get messed up, or is it more about making sure stale balances can be proved by an external party without the pod owner griefing this external party or something?

Reply:
indeed, the intermediate storage is the issue. we do multiple state updates in the `_validatorPubkeyHashToInfo` mapping. it may be possible to remove all of those and make a fully self-contained checkpoint system, but it adds a lot of complexity.